docs/ref/forms/api.txt

```diff
@@ -1499,12 +1521,57 @@ be implemented as follows::
 
 
     class GPSCoordinatesField(Field):
-        def get_bound_field(self, form, field_name):
-            return GPSCoordinatesBoundField(form, self, field_name)
+        bound_field_class = GPSCoordinatesBoundField
 
 Now you can access the country in a template with
 ``{{ form.coordinates.country }}``.
 
+You may also want to customize the default form field template rendering. For
+example, you can override :meth:`.BoundField.label_tag` to add a custom class::
+
+   class StyledLabelBoundField(BoundField):
+       def label_tag(self, contents=None, attrs=None, label_suffix=None, tag=None):
+           attrs = attrs or {}
+           attrs["class"] = "wide"
+           return super().label_tag(contents, attrs, label_suffix, tag)
+
+
+   class UserForm(forms.Form):
+       bound_field_class = StyledLabelBoundField
+       name = CharField()
+
+This would update the default form rendering:
+
+.. code-block:: pycon
+
+   >>> f = UserForm()
+   >>> print(f["name"].label_tag)
+   <label for="id_name" class="wide">Name:</label>
+
+Another requirement could be that you wanted to add a CSS class to the wrapping
+HTML element of all fields. You could easily achieve this by defining
+additional CSS classes as follows::
+
+    class WrappedBoundField(BoundField):
+        def css_classes(self, extra_classes=None):
+            return f"field {super().css_classes(extra_classes)}"
+
+
+    class UserForm(forms.Form):
+        bound_field_class = WrappedBoundField
+        name = CharField()
+
+This would update the form rendering as follows:
+
+.. code-block:: pycon
+
+   >>> f = UserForm()
+   >>> print(f["name"].as_field_group())
+   <div class="field"><label for="id_name">Name:</label><input type="text" name="name" required id="id_name"></div>
```

Comment:
Should it be mentioned that the HTML output is edited by hand to remove whitespace?

Reply:
Not necessary I'd say.